### PR TITLE
Run featurized tests on sub-packages

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Rust fails if an unsupported feature is passed to a crate and therefore the following command does not send the specified features to sub-packages:

`cargo test --all --features=bpf_c,bpf_rust,erasure,chacha,cuda --lib -- --nocapture --test-threads=1`

This means that CI has not been running any of the program featurized tests

For more information on the rust issue see: https://github.com/rust-lang/cargo/issues/5364

#### Summary of Changes

Explicitly run the tests for any package that contains features in ci/stable-perf.sh.  This is unfortunate since it requires folks to add their packages to the CI scripts if they contain features and if they don't, Cargo will quietly not run their tests :-(. Any better ideas?

Fixes #

### CODE REVIEW

**Comment by jackcmay** [comment]
@mvines See any cases I might have accidentally eliminated?

**Comment by mvines** [comment]
I’m not hot about removing all those tests from the non-perf build.  Why was that unavoidable?  I use ci/integration-tests.sh a lot in my dev workflow too, so I’m a little sad to see that go without a clear replacement for it

**Comment by jackcmay** [comment]
@mvines 
Here is a list of all the tests that `cargo test --all` runs in `test-stable.sh`.  Any missing?
```bash
     Running target/debug/deps/solana-59c704f6630c194d
     Running target/debug/deps/cluster_info-049957dc83884adf
     Running target/debug/deps/crds_gossip-0c63fc9223f38fa5
     Running target/debug/deps/fork_selection-8f247a6d6e26bf1d
     Running target/debug/deps/gossip-f36f66e27987e1f6
     Running target/debug/deps/multinode-be80a13a7f4b2974
     Running target/debug/deps/replicator-ac3240983b95b4a4
     Running target/debug/deps/rpc-907051a42607237d
     Running target/debug/deps/tvu-1507003362394693
     Running target/debug/deps/solana_bench_streamer-e400e0198e360743
     Running target/debug/deps/solana_bench_tps-638c6adb2317b248
     Running target/debug/deps/solana_bpf_loader-f6b5f65e30022f85
     Running target/debug/deps/solana_budget_program-360b3e5432e09c0d
     Running target/debug/deps/solana_drone-b5311879b9b6fa11
     Running target/debug/deps/solana_drone-be3e326c09967a2f
     Running target/debug/deps/local_drone-e82e6307bd65dba7
     Running target/debug/deps/solana_erc20-2d6310eaa93ab2d7
     Running target/debug/deps/solana_fullnode-1358ce71c146d6fb
     Running target/debug/deps/solana_fullnode_config-8f046c50f398d8b4
     Running target/debug/deps/solana_fullnode_config-5575084bee3812b0
     Running target/debug/deps/solana_genesis-1a408197485913d5
     Running target/debug/deps/solana_keygen-533733adada5dba3
     Running target/debug/deps/solana_ledger_tool-324c2bd96e46a005
     Running target/debug/deps/basic-286ad3493aa46d61
     Running target/debug/deps/solana_logger-ac0dd255b00b2ef0
     Running target/debug/deps/solana_metrics-a3ca24187a9c215d
     Running target/debug/deps/solana_native_loader-05508f677bd004f8
     Running target/debug/deps/solana_netutil-ab39c91441e9909d
     Running target/debug/deps/noop-7520b71a68ce5146
     Running target/debug/deps/programs-4f2133f914a29db9
     Running target/debug/deps/solana_replicator-7def50d1a03c4faa
     Running target/debug/deps/solana_rewards_api-8a4ae12cbe4db9b9
     Running target/debug/deps/solana_rewards_program-59cfede0fee7436c
     Running target/debug/deps/rewards-7c18c65317fe5b0a
     Running target/debug/deps/solana_runtime-78e63da19f87a20c
     Running target/debug/deps/solana_sdk-830d0e2392306ff7
     Running target/debug/deps/solana_storage_program-4d7aca4c1c8f7408
     Running target/debug/deps/storage-4df2983f8ad3df5c
     Running target/debug/deps/solana_system_program-59be48fb2c9194c1
     Running target/debug/deps/solana_upload_perf-4285af8b64cd5898
     Running target/debug/deps/solana_vote_program-315e81f95ea9df68
     Running target/debug/deps/solana_vote_signer-478cad3cd8ff042d
     Running target/debug/deps/solana_vote_signer-20712bbb7869c0af
     Running target/debug/deps/solana_wallet-2b63873b48f64119
     Running target/debug/deps/solana_wallet-009e83a1ff7c3a1b
     Running target/debug/deps/deploy-a407f3407318df3b
     Running target/debug/deps/pay-058c9bcedf24afb5
     Running target/debug/deps/request_airdrop-cc0ab17df339f58a
   Doc-tests solana_native_loader
   Doc-tests solana_logger
   Doc-tests solana_metrics
   Doc-tests solana_netutil
   Doc-tests solana_vote_signer
   Doc-tests solana_drone
   Doc-tests solana_system_program
   Doc-tests solana_bpf_loader
   Doc-tests solana-wallet
   Doc-tests solana
   Doc-tests solana_rewards_api
   Doc-tests solana_fullnode_config
   Doc-tests solana-sdk
   Doc-tests solana_runtime
```

I think `cargo test --all` does what we expect in the non-perf build before this change it looks like we were running some tests multiple times.

**Comment by mvines** [comment]
Ah, cool. 

So I'd like to see test-stable.sh invoke test-stable-perf.sh or vice versa again.  Without that link they will start to diverge again.  For example`ci/localnet-sanity.sh` is no longer run for stable-perf, and in the past we've very much seen stable-perf specific issues with this test

**Comment by jackcmay** [comment]
One bummer thing is since different crates have to be built with different features there is a new rebuild each time we switch features within the same script.  To "work around this we could maybe just push all the features to all the crates.

**Comment by jackcmay** [comment]
@mvines, How about this?

Also, would you like me to add back the integration test script as a stand-alone script?

File: ci/test-pre.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+ci/affects-files.sh \
+  .rs$ \
+  Cargo.lock$ \
+  Cargo.toml$ \
+  ci/test-stable-perf.sh \
+  ci/test-stable.sh \
+|| {
+  annotate --style info \
```
Comment (mvines) [inline ci/test-pre.sh]: `test-stable-perf.sh` we want to skip, but `test-stable.sh` should always run even on a README.md change to catch cases where we edit some random data file that has a hidden dependency

File: ci/test-stable.sh [hunk 2]
```diff
@@ -1,42 +1,10 @@
 #!/usr/bin/env bash
```
Comment (mvines) [inline ci/test-stable.sh]: How about we use the symlink trick that `gce.sh` and `ec2.sh` use: https://github.com/solana-labs/solana/blob/f0f55af35b3c723f371ace8bc85bc5335586d091/net/gce.sh#L8

test-stable-perf.sh is then a symlink to test-stable.sh, and we select what to do between the pre/post based on the filename and we avoid the test-pre.sh/test-post.sh files

**Comment by mvines** [comment]
> would you like me to add back the integration test script as a stand-alone script?
I'll get over it 😭

File: ci/test-stable.sh (hunk 2 above)
Comment (jackcmay) [reply ci/test-stable.sh]: Yeah, cool trick, didn't like the pre/post

**Pull request merged**